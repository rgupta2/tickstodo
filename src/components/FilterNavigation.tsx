import * as React from "react";
import * as styles from "./css/styles.css";
import { Filter } from "./Filter";

interface IProps {
    visibilityFilter: string;
    onFilterClick: Function;
}

export const FilterNavigation = (props: IProps) => {
    const filters: string[] = ["ALL", "ACTIVE", "COMPLETED"];

    return (
        <div className={styles.filterNav}>
            {filters.map((filter: string) => { return (
            <div className={styles.filterNavItem}>
                <Filter
                    key={filter}
                    filter={filter}
                    currentFilter={props.visibilityFilter}
                    onLinkClick={props.onFilterClick}>
                    {filter}
                </Filter>
            </div>);
        })}
        </div>);
};
