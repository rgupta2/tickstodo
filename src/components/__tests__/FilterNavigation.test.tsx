
import * as React from "react";
import * as renderer from "react-test-renderer";
import { FilterNavigation } from "../FilterNavigation";

it("FilterNavigation renders correctly", () => {
    const tree = renderer.create(
        <FilterNavigation visibilityFilter="" onFilterClick={jest.fn}/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
