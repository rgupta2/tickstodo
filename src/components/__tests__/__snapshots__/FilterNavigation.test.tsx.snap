// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterNavigation renders correctly 1`] = `
<div>
  <div>
    <a
      href="#"
      onClick={[Function]}
    >
      ALL
    </a>
  </div>
  <div>
    <a
      href="#"
      onClick={[Function]}
    >
      ACTIVE
    </a>
  </div>
  <div>
    <a
      href="#"
      onClick={[Function]}
    >
      COMPLETED
    </a>
  </div>
</div>
`;
